hw/ip/otbn/dv/rig/rig/gens/bad_branch.py

```diff
@@ -129,12 +63,12 @@ def gen(self,
         assert grs2_choices
 
         op_val_grs2 = random.choice(grs2_choices)
+        assert op_val_grs2 is not None
 
-        op_vals = [op_val_grs1, op_val_grs2, off_enc]
-
-        prog_insn = ProgInsn(chosen_insn, op_vals, None)
-
-        snippet = ProgSnippet(model.pc, [prog_insn])
-        snippet.insert_into_program(program)
+        return (op_val_grs2, chosen_insn)
 
-        return (snippet, True, model)
+    def snippet_end(self,
+                    model: Model,
+                    prog_insn: ProgInsn) -> Optional[bool]:
+        done_flag = True
+        return done_flag
```

Comment:
I don't think this variable does anything. Just `return True`?

But I'd actually suggest changing this to returning `None` and deriving `done_flag` from `self.ends_program` in the base class. Then the base class can have a no-op implementation of this function and `BadBranch` doesn't need to implement it at all. Also, maybe this could be called something like `update_for_insn` to make it a bit clearer what it does.